Assets/Scripts/Level/LevelController.cs

```diff
@@ -0,0 +1,37 @@
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using UnityAtoms;
+using UnityAtoms.FSM;
+using UniRx;
+
+public class LevelController : MonoBehaviour
+{
+    [SerializeField] private SceneDataBase sceneDB;
+    [SerializeField] private FiniteStateMachineReference levelState;
```

Comment:
```suggestion
    [SerializeField] private FiniteStateMachine levelState;
```  
I think this would be preferable, since we don't want the state machine to be kept between levels... though we still need the machine Reset(), for some reason